tests/examples/shims/screen.lua

```diff
@@ -24,8 +24,18 @@ local function create_screen(args)
         geo.height = args2.height or geo.height
     end
 
+    s.outputs = {{
+        mm_width  = 0,
+        mm_height = 0,
+    }}
+
     local wa = args.workarea_sides or 10
 
+    -- This will happen if `clear()` is called
+    if mouse and not mouse.screen then
+        mouse.screen = s
```

Comment:
Uhm... this does not work. Your `__newindex` in `mouse` will catch this instead.
So, how come you are fixing this "`mouse.screen` can refer to an already-deleted screen" problem three times and each time is half-broken?

You wrote yourself that this commit is the result of several (old?) commits that were through many rounds of rebasing and were just now squashed. Perhaps it makes sense to rewrite this commit in sensible, small parts guided by what is actually needed by some of the doctests?

Reply:
> Perhaps it makes sense to rewrite this commit in sensible, small parts guided by what is actually needed by some of the doctests?

That was the original idea, however I gave up because I have a "main branch" where all my PRs (submitted or yet to be submitted) are to be able to dogfood everything at once. However it got to the point I could no longer " `rebase -i`+remove the old merge commit and merge the PR again" without breaking everything.

This is in part because I have too many branches and too many ideas half implemented, but it is also because I am an active AwesomeWM developer and it will inevitably happen. Not all ideas are good ideas and not all ideas are worth spending time to get the past your review process ;).

So in the end I squashed this and now try to get it merged all at once so I can stop wasting time rebasing conflicts every single time I update one of the PR.

It ain't perfect and I prefer smaller commits, but it is no longer manageable with a day job, hobbies, outdoor activities and life itself. Better go though the review now and get over it. Now that the "upstream" shims are more complete, then I can do smaller changes to them again with decreased risk of rebase issues.